### DISCUSSION TITLE
make it so fields can be added to the end?

### DISCUSSION THREAD

**Discussion by prescriptionlifeline** [Laravel-Backpack/community-forum#1019] [Ideas]
Per https://backpackforlaravel.com/docs/6.x/crud-api#create-update-operations you can add a field after another one or before another one but you can't add one at the end.

If you want to add a field at the beginning then you can just use `->first()` for that column in that migration and then call `CRUD::setFromDb()` but if you want to add a field at the end your option are:

1. Have the migration add the new column after what's currently the last column but then if you add new columns you'll have to have the new migration reset the position of the column that you want to be the last one
2. You could just specify all the columns in `setup*Operation()` instead of using `CRUD::setFromDb()`

Between those two options I guess the second is preferable but I'd prefer something like `$this->crud->addField()->placementFromEnd(0)` or some such instead.

**Comment by pxpm** [comment]
@prescriptionlifeline indeed, `setFromDb()` is not the "best" thing to use. 

I never, ever use it and never recommend anyone to use it.
In fact we already have on the table the possibility of completely deprecated it, in favor of something that actually generates the fields in your CRUD operations (similar to **2.** you mentioned), maybe part of `DevTools` or `generators` .

For now I really advise you to use option 2. If you have multiple CRUDS that has similar/same fields you can always create a Trait with those field definitions and add it to your controllers. 

I am going to close this, but if you feel I am wrong, please re-open, or comment back and we can re-open it, and evaluate the situation. 

Cheers 🙏
